core/msg.c

```diff
@@ -93,10 +95,12 @@ static int _msg_send(msg_t *m, kernel_pid_t target_pid, bool block, unsigned sta
         return -1;
     }
 
+    thread_t *me = (thread_t *) sched_active_thread;
```

Comment:
sched_active_thread is marked volatile. Copying the value to "me" allows holding it in a register, instead of having to load it from memory every time. Also, it saves the otherwise necessary casting. (sched_active_thread might even change, but never while this thread is executing).